src/Adapter/Order/Refund/OrderRefundCalculator.php

```diff
@@ -82,48 +84,56 @@ public function computeOrderRefund(
-            $refundedAmount += $productRefund['amount'];
+            $refundedAmount = $refundedAmount->plus(new Number((string) $productRefund['amount']));
         }
 
         $voucherChosen = false;
-        $voucherAmount = 0;
+        $voucherAmount = new Number('0');
         if ($voucherRefundType === VoucherRefundType::PRODUCT_PRICES_EXCLUDING_VOUCHER_REFUND) {
-            //@todo: Check if it matches order_discount_price in legacy
-            $refundedAmount -= $voucherAmount = (float) $order->total_discounts;
+            $voucherAmount = new Number((string) $order->total_discounts);
+            $refundedAmount = $refundedAmount->minus($voucherAmount);
         } elseif ($voucherRefundType === VoucherRefundType::SPECIFIC_AMOUNT_REFUND) {
             $voucherChosen = true;
             $refundedAmount = $voucherAmount = $chosenVoucherAmount;
         }
 
-        $shippingCostAmount = $shippingRefund ?: false;
-        if ($shippingCostAmount > 0) {
-            $shippingMaxRefund = $isTaxIncluded ? $order->total_shipping_tax_incl : $order->total_shipping_tax_excl;
+        $shippingCostAmount = new Number((string) ($shippingRefund ?? 0));
+        if ($shippingCostAmount->isPositive()) {
+            $shippingMaxRefund = new Number(
+                $isTaxIncluded ?
+                    (string) $order->total_shipping_tax_incl :
+                    (string) $order->total_shipping_tax_excl
+            );
+
             $shippingSlipResume = OrderSlip::getShippingSlipResume($order->id);
-            $shippingMaxRefund -= $shippingSlipResume['total_shipping_tax_incl'] ?? 0;
-            if ($shippingCostAmount > $shippingMaxRefund) {
+            $shippingSlipTotalTaxIncl = new Number((string) ($shippingSlipResume['total_shipping_tax_incl'] ?? 0));
+            $shippingMaxRefund = $shippingMaxRefund->minus($shippingSlipTotalTaxIncl);
+
+            if ($shippingCostAmount->isGreaterThan($shippingMaxRefund)) {
                 $shippingCostAmount = $shippingMaxRefund;
             }
             if (!$isTaxIncluded) {
-                // @todo: use https://github.com/PrestaShop/decimal for price computations
-                $refundedAmount += $taxCalculator->addTaxes($shippingCostAmount);
+                $taxesAmount = $taxCalculator->addTaxes((float) $shippingCostAmount->toPrecision(CommonAbstractType::PRESTASHOP_DECIMALS));
+                $taxes = new Number((string) $taxesAmount);
+                $refundedAmount = $refundedAmount->plus($taxes);
             } else {
-                $refundedAmount += $shippingCostAmount;
+                $refundedAmount = $refundedAmount->plus($shippingCostAmount);
             }
         }
 
         // Something has to be refunded (check refunds count instead of the sum in case a voucher is implied)
-        if (count($productRefunds) <= 0 && $shippingCostAmount <= 0) {
+        if (count($productRefunds) <= 0 && $refundedAmount->isLowerOrEqualThan(new Number('0'))) {
```

Comment:
I believe @atomiix also needed to perform these kind of checks, it's a pity that we need to instanciate a `new Number('0')` to perform this check Maybe we could add a new method to the `Number` class but it requires updating our `prestashop/decimal` library
@eternoendless do you think it could be possible?

Reply:
then maybe we could also implement toFloat() ? because its still needed for some legacy class computations. And now it seems possible to do (float) (string) $number which looks strange

Reply:
@jolelievre @zuk3975 I agree, doing `new Number('0')` for each check is a pity... As we'll be using more this library I think it would be useful to have methods to make that easier.

In the other hand, the need for a `toFloat()` method will be less useful for the same reasons: if we use `Number` everywhere we will (theoretically 😄) not need to cast to float anymore.

Reply:
Well in the long term when everything uses `Number` you might be right, but it would still be more handy to use than this ugly `(float) (string)` ... And since the purpose of this class is to deal with decimal numbers (aka float) I don't think it's shocking to allow an output as a float (but it's just my opinion)
However having a static method to build from a float would be even more handy!! Along with an helper to compare to zero which is a very specific and used value

I'd say we could add a const `Number::ZERO` which would contains the 0 value in a number would be ideal, but unless I'm mistaken, we can't use objects as const So maybe a static function? with a static (🙀) var to avoid instanciating the object a thousand time.

Reply:
@jolelievre Create an issue in PrestaShop/decimal and we can discuss there.

Reply:
Hop, here's the issue https://github.com/PrestaShop/decimal/issues/6